app/uk/gov/hmrc/economiccrimelevyreturns/controllers/CancelReturnAmendmentController.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyreturns.controllers
+
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.economiccrimelevyreturns.config.AppConfig
+import uk.gov.hmrc.economiccrimelevyreturns.controllers.actions.{AuthorisedAction, DataRetrievalAction}
+import uk.gov.hmrc.economiccrimelevyreturns.forms.CancelReturnAmendmentFormProvider
+import uk.gov.hmrc.economiccrimelevyreturns.forms.FormImplicits._
+import uk.gov.hmrc.economiccrimelevyreturns.services.ReturnsService
+import uk.gov.hmrc.economiccrimelevyreturns.views.html.CancelReturnAmendmentView
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class CancelReturnAmendmentController @Inject() (
+  val controllerComponents: MessagesControllerComponents,
+  authorise: AuthorisedAction,
+  getReturnData: DataRetrievalAction,
+  returnsService: ReturnsService,
+  formProvider: CancelReturnAmendmentFormProvider,
+  appConfig: AppConfig,
+  view: CancelReturnAmendmentView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport
+    with BaseController
+    with ErrorHandler {
+
+  val form: Form[Boolean] = formProvider()
+
+  def onPageLoad(): Action[AnyContent] = (authorise andThen getReturnData) { implicit request =>
+    Ok(view(form.prepare(None), request.startAmendUrl))
+  }
+
+  def onSubmit(): Action[AnyContent] = (authorise andThen getReturnData).async { implicit request =>
+    form
+      .bindFromRequest()
+      .fold(
+        formWithErrors => Future.successful(BadRequest(view(formWithErrors, request.startAmendUrl))),
+        cancelReturnAmendment =>
+          if (cancelReturnAmendment) {
+            (for {
+              _ <- returnsService.deleteReturn(request.eclReturn.internalId).asResponseError
+            } yield ()).fold(
+              _ => Redirect(routes.NotableErrorController.answersAreInvalid()),
```

Comment:
If the deletion fails, it's not answersAreInvalid that needs to be returned.

We now have routeError, see example: https://github.com/hmrc/economic-crime-levy-returns-frontend/blob/f5adf3b9a87fbbc5b8c1ceca9143656bd3a2a536/app/uk/gov/hmrc/economiccrimelevyreturns/controllers/AmendReasonController.scala#L67

Reply:
Fixed.